### DISCUSSION TITLE
Menu with link and subMenu

### DISCUSSION THREAD

**Discussion by lento4ka** [adobe/react-spectrum#3620] [Q&A]
Hi, could you tell how to make a menu with elements like:
1. items
2. sections
3. links
4. subMenu that contains all the same menu items

There are no necessary examples in the documentation.

Now use `useMenuTriggerState, useMenu, useTreeState, useMenuItem`.

**Comment by snowystinger** [comment]
We haven't implemented submenu's yet, but this is the pattern we'd follow to do it. https://www.w3.org/WAI/ARIA/apg/example-index/menubar/menubar-navigation
I don't think any of us have looked to see how much work is involved in supporting this recently. Any docs we would have on it would be pretty old. We have it in our backlog to pick up again in the future.
